api/src/pcapi/core/educational/api/stock.py

```diff
@@ -83,7 +84,22 @@ def create_collective_stock(
 def edit_collective_stock(
     stock: educational_models.CollectiveStock, stock_data: dict
 ) -> educational_models.CollectiveStock:
-    validation.check_if_offer_is_not_public_api(stock.collectiveOffer)
+    if not feature.FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active():
+        # with FF ON this is included in the allowed actions
+        validation.check_if_offer_is_not_public_api(stock.collectiveOffer)
+
+    is_editing_dates = (
+        "beginningDatetime" in stock_data
+        or "startDatetime" in stock_data
+        or "endDatetime" in stock_data
+        or "bookingLimitDatetime" in stock_data
+    )
```

Comment:
```suggestion
    dates_attributes = ("beginningDatetime", "startDatetime", "endDatetime", "bookingLimitDatetime")
    is_editing_dates = any(attribut in stock_date for attribut in dates_attributes)
```

et on peut potentiellement réutiliser le dates_attributes

Reply:
Why not pout le any(...) mais où on réutiliserait dates_attributes ?

Reply:
pas d'idée, mais c'est un concept qui peut avoir du sens ailleurs. Peut-être a mettre sur le model ? ou pas..

Reply:
allez je prends